### PR TITLE
Add content regarding Astropy project roles

### PR DESCRIPTION
**Pull request by taldcroft**
The Astropy project will now formally identify project roles and responsible parties. This includes responsibilities ranging from direct package development to communication, distribution, and managerial activities.

This PR adds:

- New section to `about.html`
- New file `roles.txt` which defines role leads and deputies.
- New script `update_roles.py` that updates `about.html` based on `roles.txt`.
- New sub-page `role_responsibilities.html` that describes each role in detail.

See: http://hea-www.harvard.edu/~aldcroft/tmp/astropy.github.com/about.html#roles

@astrofrog @eteq @kelle

### CODE REVIEW

**Comment by kelle** [comment]
add some text to the Community engagement coordinator section. 
- Add links to twitter, and FB group. https://www.facebook.com/groups/astropython/, 
- add 'Facilitate activities which foster the engagement of new contributors to the project.'

**Comment by eteq** [comment]
Roles that should be "willing to step down" for me:
* Astropy.org web page maintainer
* `astropy.utils` maintainer
* release coordinator

Another suggested change:
* remove Tom A as a deputy for `astropy.coordinates`, add Marten and Stuart Littlefair as deputies

**Comment by astrofrog** [comment]
> release coordinator

If we find someone else who wants to help, then I'd be happy to step up to lead for this role while they are trained as a deputy.

**Comment by astrofrog** [comment]
Things I would be 'willing to step down from':

* lead tweep (Twitter)
* MacPorts maintainer (Moritz expressed interest in helping for this, I'm trying to convince him to become lead)

I'd like to be changed to deputy for the following:

* ``astropy.io.misc``
* ``astropy.vo.samp``
* ``astropy.visualization``
* ``package-template``

But not sure if it makes sense to change that yet, or whether it would actually intimidate people (who might be less enclined to volunteer for lead than deputy)

**Comment by eteq** [comment]
Oh, to be clear, all of the things I am "willing to step down from" are probably better phrased as "could be changed to deputy", as it would still make sense for me to be involved at a lower level

**Comment by taldcroft** [comment]
Sorry for the long delay.  I *think* I have put in the necessary changes, though some of the inputs were slightly ambiguous.  But now it should be easier to look at the final output and suggest any additional deltas.

Once this is approved and merged then we can advertise this on wider channels.

File: roles.txt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+Astropy.org web page maintainer | | Erik Tollerud::Would prefer deputy role | UNFILLED
+
+Astropy-helpers maintainer | | UNFILLED | Erik Tollerud, Brigitta Sipocz
+
+Astropy GSoC coordinator | 2016 | Tom Aldcroft | Erik Tollerud
+
+CI-helpers maintainer | | Brigitta Sipocz |  UNFILLED
+
+Community engagement coordinator | Overall  | Kelle Cruz | UNFILLED
+                                 | Twitter  | Tom Robitaille::Would prefer deputy role | Adrian Price-Whelan
```
Comment (adrn) [inline roles.txt]: Ah, I see now @astrofrog wants to be deputy -- you can swap us and put me as lead.

Comment (taldcroft) [inline roles.txt]: @adrn - I just made that change and re-ran the script (which I have run dozens of times already) to regenerate the HTML, and get:
```
neptune$ ./update_roles.py 
python(8157,0x7fff78210310) malloc: *** error for object 0x10055bc01: pointer being freed was not allocated
*** set a breakpoint in malloc_error_break to debug
Abort trap: 6
neptune$ ./update_roles.py 
Segmentation fault: 11
```
Looks like you cannot be twitter coordinator, software doesn't allow it.  :smile:

**Comment by taldcroft** [comment]
@adrn - OK, twitter role swap done.  I had malformed the input, but this is a real bug (https://github.com/astropy/astropy/issues/5219).

**Pull request merged**